controller/src/test/java/io/pravega/controller/store/stream/TableHelperTest.java

```diff
@@ -142,8 +143,93 @@ public void getActiveSegmentsTest() {
         assertEquals(newSegments, activeSegments);
     }
 
+    @Test
+    public void rollingTxnTest() {
```

Comment:
This is a good test, but I'm wondering if it is worth testing the manipulation of epochs separately, in a more unit test fashion to verify that we have the correct behavior independent of txns.

Reply:
manipulation of epochs is tested in several other tests.
This test verifies response of other table queries if there are partial state updates.
ControllerEventProcessorTest tests complete flow of commitrequesthandler.
ScaleRequestHandlerTest tests transaction's commit relationship wrt scale operations and verifies rolling transaction takes place whenever necessary.
StreamMetadataStore test takes an inside view of the processing and tests running different steps of rolling transaction multiple times. 
Apart from that, EndToEndAutoScaleWithTxnTest verifies that rolling transaction takes place in a complete deployment of segmentstore and controller and real clients.

Reply:
We can add more unit testing esp for any specific scenarios we want to verify and harden.